plugins/vars/sops.py

```diff
@@ -0,0 +1,112 @@
+from ansible.plugins.vars import BaseVarsPlugin
+from ansible.inventory.host import Host
+from ansible.inventory.group import Group
+from ansible.utils.vars import combine_vars
+from ansible_collections.community.sops.plugins.module_utils.sops import Sops, SopsError
+
+from ansible.utils.display import Display
+display = Display()
+
+
+DOCUMENTATION = '''
+    vars: sops_vars
+    author: Edoardo Tenani (@endorama) <e.tenani@arduino.cc>
+    short_description: Loading sops-encrypted vars files
+    version_added: "2.10"
+    description:
+        - Load encrypted YAML files into corresponding groups/hosts in group_vars/ and host_vars/ directories.
+        - Files are encrypted prior to reading, making this plugin an effective companion to host_group_vars plugin.
+        - Files are restricted to .sops.yaml, .sops.yml, .sops.json extensions.
+        - Hidden files are ignored.
+    options:
+      _valid_extensions:
+        default: [".sops.yml", ".sops.yaml", ".sops.json"]
+        description:
+          - "Check all of these extensions when looking for 'variable' files which should be YAML or JSON or vaulted versions of these."
+          - 'This affects vars_files, include_vars, inventory and vars plugins among others.'
+        type: list
+'''
+
+
+FOUND = {}
+DEFAULT_VALID_EXTENSIONS = [".sops.yaml", ".sops.yml", ".sops.json"]
+
+
+class VarsModule(BaseVarsPlugin):
+
+    def get_vars(self, loader, path, entities, cache=True):
+        ''' parses the inventory file '''
+
+        if not isinstance(entities, list):
+            entities = [entities]
+
+        super(VarsModule, self).get_vars(loader, path, entities)
+
+        data = {}
+        for entity in entities:
+            if isinstance(entity, Host):
+                subdir = 'host_vars'
+            elif isinstance(entity, Group):
+                subdir = 'group_vars'
+            else:
+                raise AnsibleParserError("Supplied entity must be Host or Group, got %s instead" % (type(entity)))
+
+            # avoid 'chroot' type inventory hostnames /path/to/chroot
+            if not entity.name.startswith(os.path.sep):
+                try:
+                    found_files = []
+                    # load vars
+                    b_opath = os.path.realpath(to_bytes(os.path.join(self._basedir, subdir)))
+                    opath = to_text(b_opath)
+                    key = '%s.%s' % (entity.name, opath)
+                    self._display.vvvv("key: %s" % (key))
+                    if cache and key in FOUND:
+                        found_files = FOUND[key]
+                    else:
+                        # no need to do much if path does not exist for basedir
+                        if os.path.exists(b_opath):
+                            if os.path.isdir(b_opath):
+                                self._display.debug("\tprocessing dir %s" % opath)
+                                found_files = loader.find_vars_files(opath, entity.name)
```

Comment:
This doesn't appear to find files .sops.{yaml,yml,json} vars files when I run it via a local vars_plugins. Passing `extensions=DEFAULT_VALID_EXTENSIONS` to `loader.find_vars_files` does work, though, and seems to allow skipping the second filtering of the list (as in the diff below). Not sure if this is a compatibility concern for some versions of Ansible since I only tested this on Ansible 2.9.10, so grain of salt on that.

```suggestion
                                found_files = loader.find_vars_files(opath, entity.name, extensions=DEFAULT_VALID_EXTENSIONS)
```

Reply:
Thanks for the input, I changed the code as suggested. (I need your email to add you as Co-author, if you're interested just let me know which email should I use).

Reply:
ncower@nil.dev is fine (the one on my GitHub profile / commits, if that's visible anymore), though it's probably a bit of a leap to say 'co-author'. Contributor or person-who-made-one-comment is fine.